### ISSUE TITLE
Materialize as Read Replica

### ISSUE THREAD

**Issue by rjnn** [MaterializeInc/materialize#1968]
This is an epic to hold all issues related to using Materialize as a full-fledged replica for MySQL/Postgres. Supporting this use-case requires that we solve some Quality-of-Life issues (read the schema changes topic to drop/create views) as well as some thorny problems that might require changes to Debezium itself (propagating timestamp/progress information through Debezium).

This epic is scoped explicitly to using Debezium only. Other replication frameworks are out. It's an open question whether we want to support `tb` (this involves actually improving `tb` to the point where it's usable), or whether we require the stack to look liked Mysql/Postgres -> Debezium -> Kafka Connect -> Kafka -> Materialize.

### Task List
- [ ] Automate setting up debezium with Postgres #1282 
- [ ] DROP DATABASE in MySQL is ignored #555 
- [ ] Determine how TimestampTZs come out of debezium #854 
- [ ] Convert from debezium-provided TimestampTZs to UTC #856

**Comment by cuongdo** [comment]
- [ ] Support propagating MySQL/Debezium DROP TABLE to Materialize #1965
- [ ] Automate setting up debezium with Postgres #1282 
- [ ] DROP DATABASE in MySQL is ignored #555 
- [ ] Determine how TimestampTZs come out of debezium #854 
- [ ] Convert from debezium-provided TimestampTZs to UTC #856

(Converted from ZenHub epic)

**Comment by elindsey** [comment]
@awang has our thinking on this changed now that we have ways of doing direct replication?

It seems like we may still need to support a few things for correctness when using dbz (such as reading the schema change topic), but I'm not sure how 'being a read replica when using dbz' differs from 'have good kafka dbz sources', and if it's still relevant in light of direct db cdc.

**Comment by awang** [comment]
TL;DR: I don't have a clean answer for you yet, unfortunately -- figuring out how to prioritize Debezium support is one we're going to have to continue to discuss.

Direct replication is amazing for quickly getting started, but we don't anticipate it will work for everyone with the current architecture, because it [tightly couples](https://materialize.com/docs/sql/create-source/postgres/#postgres-replication-slots) Materialize and the database.  If users want a more fault-tolerant architecture, I think we have a few options:

1. Add some sort of fault-tolerance to our direct-replication architecture.  Just spitballing, but this might involve keeping Materialize close to the source database, and writing to a Kafka sink, which could then be consumed by a downstream Materialize that does more work. But then, that starts to sound like Debezium....
2. Improve our Debezium support
3. Integrate with hosted CDC providers

(1) doesn't seem like a great idea.
(3) is a near-term priority -- it's not a strict "Materialize 1.0" necessity, but one we'd like to include if possible.   I _think_ (3) is more urgent than (2), but we still don't have a great sense of where it sits relative to all the other global priorities

**Comment by elindsey** [comment]
Gotcha, that makes sense. 

I spent a little while looking through what Debezium offers in terms of DDL propagation. It seems like the closer we get to full read replica (eg. DDL propagation like processing drops, creates, etc.), the leakier Debezium gets as an abstraction - more inconsistencies start showing up depending upon which upstream database it's attached to. It's possible that we'll end up not talking about Debezium, but "Debezium when rigged up to MySQL", "Debezium when rigged up to Postgres", etc., implication being that guidance for how to scale past direct CDC replication may vary based on the database and on the eng side we might need to do bespoke work per db we want to support through dbz.

Nothing actionable or pressing, just got curious about this today. 🙂

**Comment by elindsey** [comment]
There's no longer value in this top level tracking task
